src/main/java/teammates/ui/webapi/output/CommonSearchLinksData.java

```diff
@@ -0,0 +1,52 @@
+package teammates.ui.webapi.output;
+
+/**
+ * Contains the common links for instructors and students.
+ */
+public class CommonSearchLinksData {
+    protected boolean showLinks;
+    protected String email;
+    protected String manageAccountLink;
+    protected String homePageLink;
+    protected String courseJoinLink;
```

Comment:
I thought we have agreed that the link construction should be done in the frontend?

Reply:
Yea, but in the process of migration, we felt like constructing the links required a lot of work. (generating many hashsets and hashmaps). If this was sent over to the frontend, it would bloat the response object.

Also, also some of the links (courseJoinLink for example) are keys - that data isn't even available on frontend unless we want to send the keys over. Some of these keys look to be encrypted so it might also not be good to send it over

Reply:
> required a lot of work. (generating many hashsets and hashmaps)

In many case, you don't need the hashmaps or hashsets. They can usually be represented in the form of a list.

>  that data isn't even available on frontend unless we want to send the keys over.

This is not a big deal. When you send the link, you also send the key right? In addition, in our APP, we care most about application security. The network security is handled by HTTPS.

Reply:
I think the point on HTTPS security is fair. However, I feel that leaving front end to generate the links i kind of misusing the front end. The way I see it, the front end should primarily concern itself with rendering/doing the minimal work to render the data. Since generating the links constitute computational work, we could leave it to be done in the backend side instead of cluttering the front-end code. Let me know what you think.

Reply:
Why do you consider it as `computational work`? It looks like just string concatenation.

Reply:
Sorry, perhaps I should have made myself clearer. Yea, its string concatenation. However, this joining of links on the front end can be moved to the backend without any consequences. Furthermore, this would constitute a clearer separation of concerns. The front end shouldn't worry about the contents of the fields of the response object. It should just worry about which fields of the response object are relevant to the page and modify the front end state accordingly to render the necessary.

Reply:
Let me put it in another way. I think if you consider the modern architecture of separations of the frontend and backend. Most likely, the backend does not know the existence of the frontend. Therefore:

- it is best practice to put the presentation of data in the frontend (The frontend receives `accountId` and it decides to present it as a link. It might decide it to present it in another way like to display it)

- Consider next time when you add a new page in the frontend. Will you go back and change the backend?

Reply:
I think ideally, all the resources the front end needs should already be exposed by the backend. 
However, I don't think this is just a matter of data presentation. Looking at the link generation code, we need many specific values from files like `Config` and `Const`. We could always send this values to the frontend, however. wouldn't it just make more sense to compute them on the backend, where the values are readily available, instead of copying these values to the frontend?

Reply:
> many specific values from files like `Const`

The reason why we have `Const.WebPageURIs` is because the backend need to send emails to the users so we need to insert the link in the email. Ideally, we should keep this minimal.

> many specific values from files like `Config`

The frontend does need to know its URL. All navigate is done using `NavigationService` or `Router`.

Reply:
Yea, but if we do this work on the front end, that leaves us with 2 options.
1. We send these constants to the front end, at the risk of bloating up the response object. 
2. We hardcode these constants in the front end. However, if we do so, everytime the Backend const value changes, we have to do the same on the front end, which can be troublesome.

Reply:
Constants? For example?

Reply:
This part touches the `Const` and `Config` objects, which are only available on the backend. We could move it over to the frontend, but it has the two problems in the above comment
```
            instructorData.setManageAccountLink(Config.getFrontEndAppUrl(Const.WebPageURIs.ADMIN_ACCOUNTS_PAGE)
                    .withInstructorId(instructor.googleId)
                    .toString());
            instructorData.setHomePageLink(Config.getFrontEndAppUrl(Const.WebPageURIs.INSTRUCTOR_HOME_PAGE)
                    .withUserId(instructor.googleId)
                    .toAbsoluteString());
            instructorData.setCourseJoinLink(Config.getFrontEndAppUrl(Const.WebPageURIs.JOIN_PAGE)
                    .withRegistrationKey(StringHelper.encrypt(instructor.key))
                    .withParam(Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR)
                    .toAbsoluteString());

```

Reply:
As I have mentioned:

You don't need `Config.getFrontEndAppUrl` and `Const.WebPageURIs.*`

For `paramsName`, right now it is not standardized. You can just use string literal.

Reply:
Okay will fix